book/02-git-basics/sections/recording-changes.asc

```diff
@@ -1,26 +1,38 @@
+- repository: aslında kelime olarak depo, ambar gibi anlamlara geliyor. Ben proje kodlarınızı kaydettiğiniz bir veri ambarı olarak düşünebilirsiniz. Ben ingilizcedeki repository kelimesine karşılık olarak "repo" kısaltmasını kullanacağım.
 
-Remember that each file in your working directory can be in one of two states: _tracked_ or _untracked_.
-Tracked files are files that were in the last snapshot; they can be unmodified, modified, or staged.
-In short, tracked files are files that Git knows about.
+- Initialize: anlam olarak bir ilk durumu ifade etmektedir. Mesela bir yarış düşünelim. Herkesin başlangıç çizgisinde hizaya girmesini "initialize" olarak adlandırabiliriz. Yarış henüz başlamamıştır fakat başlaması için herşey hazırdır. Bu bağlamda `git init` komutu bir projeye başlamak için "sıfır noktası" oluşturulmasını temsil eder.
 
-Untracked files are everything else -- any files in your working directory that were not in your last snapshot and are not in your staging area.
-When you first clone a repository, all of your files will be tracked and unmodified because Git just checked them out and you haven't edited anything.
+- Stage: ingilizcede sahne anlamına gelir. Mesela farklı perdelerden (sahnelerden) oluşan bir tiyatro oyunu düşünelim. Oyunun her bir sahnesi için farklı bir arka plan ve dekorasyon (mesela ev, köy meydanı, orman vs) bulunur. Perde kapandığında sahneye yeni dekor ürünleri eklenir ve perde açıldığında oyun yeni sahnede devam eder. Git'te çalışma ağacınıza yeni bir dosya eklerseniz bu dekorun değiştiği anlamına gelir. Haliyle bu dosyayı sahneye yerleştirmelisiniz. Buna "staging" denir. "stage"e almadığınız dosyadaki değişiklikler takip edilmez. `git add <dosya>` ile yeni eklediğiniz dosyayı "stage"e alırsınız ve bu dosyadaki kod değişiklikleri Git tarafından izlenmeye başlar. Fakat bu izleme sadece sizin makineniz için geçerlidir. Bu değişiklikler "commit" geçmişinde görülmez.
 
-As you edit files, Git sees them as modified, because you've changed them since your last commit.
-As you work, you selectively stage these modified files and then commit all those staged changes, and the cycle repeats.
+-Commit: adamak veya taahhüt etmek anlamlarına gelir. Bana kalırsa Türkçede bu işlemle yapılan işi en iyi açıklayan kelime kaydetmek ifadesi. Git kullanırken yaptığınız değişiklikler sadece sizin yerel bilgisayarınızda izlenmektedir. Bu değişikliklerin uzak repoda yer alması ve herkesçe görülebilecek şekilde kayıt altına alınması için "commit" komutunu kullanmanız gereklidir. `git commit -m "kayıt_mesajı"` komutuyla, Git çalıştığınız daldaki kodun en son halinin anlık bir resmini çeker ve bunu uzak repoya kaydeder. Böylece kodunuz ileride geri dönebilmeniz için veya aynı projede farklı kodlarda çalışan arkadaşlarınızın kodun sizin çalıştığınız kısmındaki en güncel haline ulaşabilmeleri için erişime açılmış olur.
 
```

Comment:
Editorun notu gibi bir kisim mi hayal ettiniz burada?

"bu bolumu eklememin sebebi", "ben xyz yapacagim", "bana kalirsa" gibi kullanimlar bu ceviriye ne kadar uygun emin degilim.

Reply:
Kitap mevcut haliyle Türk okurunun (ingilizcesi yetersiz başlangıç seviyesi Türk yazılımcının) ihtiyacını karşılamaktan uzak. İnitialize, stage, commit gibi kelimelerin Türkçe karşılığı yok. Amacım bire bir çeviri yapmak değil, Türklere ihtiyaç duydukları bilgiyi vermek. Kitabin yazarı da kitabı para kazanmak için değil, yazılımcıları aydınlatmak için yazmış ve insanlığın istifadesine sunmuş. Benim tek odak noktam bu: İnsanlar okusun ve anlasınlar. Yaklaşım itibariyle çevirmen değil, open-source contributor olarak görüyorum kendimi.

Yetki sizde, ister yazdığım gibi kullanın, ister değiştirin, isterseniz de kitabın orjinalinde bu yok deyip atın. Ya da isterseniz şöyle yapsak nasıl olur diye beraber tartışalım.

Reply:
Yetki bende degil, benim fazladan yapabildigim tek sey PR uygunsa ve ceviri kabul gorduyse merge butonuna basmak. Yoksa kitabin yayin haklari ya da iceriginin ne olup ne olmayacagina karar vermek benim inisiyatifimde degil kesinlikle. Yazarin agzindan yazilmis gibi gorundugu icin sorma geregi duydum amaciniz neydi diye. 

Sahsi fikrim boyle bir aciklamaya gerek olmamasi yonunde, fakat kelimelerin Turkce karsiliklarinin anlami karsilamadigi konusunda katiliyorum. Initialise = baslatmak, stage = hazirlamak, commit = yapmak/islemek desek bile ek aciklama olmadan ilk seferde anlasilmasi zor. Umit ediyorum ki kitabi bastan sona okuyan birisi, kitap boyunca devam eden aciklamalardan bu icerigi ve anlami yakalayacaktir, ekstra aciklamaya bu yuzden gerek var mi bilmiyorum. 

Diger yonden, basta da belirttigim gibi, yazarin agzindan yazilmis gibi olmasi beni tetikleyen sey oldu.

Reply:
Siz nasıl uygun görüyorsanız öyle yapın hocam. Başlangıç kısmını özellikle en sona bıraktım ve ayrı bir commit olarak ekledim.